lib/ukrand/chacha.c

```diff
@@ -130,3 +149,53 @@ __u32 uk_swrand_randr_r(struct uk_swrand *r)
 		r->k = 0;
 	}
 }
+
+/* Uses the pre-initialized default generator  */
+/* TODO: Add assertion when we can test if we are in interrupt context */
+/* TODO: Revisit with multi-CPU support */
+__u32 uk_chacha_randr_u32(void)
+{
+	unsigned long iflags;
+	__u32 ret;
+
+	iflags = ukplat_lcpu_save_irqf();
+	ret = uk_chacha_randr_r(&uk_chacha_def);
+	ukplat_lcpu_restore_irqf(iflags);
+
+	return ret;
+}
+
+__u64 uk_chacha_randr_u64(void)
+{
+	unsigned long iflags;
+	__u64 ret;
+
+	iflags = ukplat_lcpu_save_irqf();
+	ret = uk_chacha_randr_r(&uk_chacha_def);
+	ret = (ret << 32) | uk_chacha_randr_r(&uk_chacha_def);
+	ukplat_lcpu_restore_irqf(iflags);
+
+	return ret;
+}
+
+ssize_t uk_chacha_fill_buffer(void *buf, size_t buflen)
```

Comment:
Try to use our own defined types where possible: `__ssz`, `__sz`.

Reply:
This was changed in a later commit, when the function was renamed to `uk_random_fill_buffer`.